MediaBrowser.Server.Implementations/FileOrganization/EpisodeFileOrganizer.cs

```diff
@@ -157,7 +157,43 @@ public Task<FileOrganizationResult> OrganizeEpisodeFile(string path, Cancellatio
         {
             var result = _organizationService.GetResult(request.ResultId);
 
-            var series = (Series)_libraryManager.GetItemById(new Guid(request.SeriesId));
+            Series series = null;
+
+            if (request.NewSeriesProviderIdsDictionary.Count > 0)
```

Comment:
Better but still a bit quirky to have a property suffixed with Dictionary. Can that be removed?

What you have now is going to create complicated post query strings that will be difficult to debug. Make the request object property a Dictionary as well, then switch the client-side post to use a json post body.  See updateServerConfiguration for an example.